src/Integration/Telemetry/Payload/ServerNotifications.cs

```diff
@@ -28,16 +31,16 @@ public sealed class ServerNotifications
         public bool IsDisabled { get; set; }
 
         [JsonProperty("count_by_type")]
-        public ServerNotificationCounters ServerNotificationCounters { get; set; } = new ServerNotificationCounters();
-    }
-
-    public sealed class ServerNotificationCounters
-    {
-        [JsonProperty("QUALITY_GATE")]
-        public ServerNotificationCounter QualityGateNotificationCounter { get; set; } = new ServerNotificationCounter();
+        [XmlIgnore]
+        public Dictionary<string, ServerNotificationCounter>  ServerNotificationCounters { get; set; } = new Dictionary<string, ServerNotificationCounter>();
```

Comment:
My bad, this should've been a dictionary and not a concrete class.

Reply:
Why? The json sent to the server was in the correct format, wasn't it?

Reply:
Still is.
This allows us not to know about the specific notification types and support all categories without needed changes.

Reply:
Do we have any other categories of notification?

It seems weird to add a strongly-typed `IServerNotificationsTelemetryManager` interface just for server notifications, then make it a "weak" wrapper round a property bag by allowing anything as a category.

Reply:
There's a difference between knowing that there are notification types and knowing exactly which types exist on the server. We don't care anywhere else in the code about the specific types, which made the strongly typed structure extremely awkward to use. 
We receive generic notifications from the server and display them as-is -- we should continue with this generic approach in collecting the telemetry as well. Otherwise we're just creating ourselves a tech debt to support any changes made on the server. 
You can take a look at the usage, the new generic structure makes it very simple: https://github.com/SonarSource/sonarlint-visualstudio/pull/2175/files#diff-4f799163db4bdb918950a9a163e5e53202bf34c3ecc4551342c81bf3144f14e9

Reply:
Making the internal data structure used by the server notifications flexible is one thing - that's an internal implementation issue. Exposing it to consumers in a weakly-typed way is another.

Reply:
Ok, makes sense. 

The original review comment needed a bit more explanation than "this should have been a dictionary".

Reply:
Fair point; I'll explain more next time.